Add configuration option for origin loop

strehle opened the pull request:
Depends on PR https://github.com/cloudfoundry/uaa/pull/2932

strehle committed:
Add configuration option for origin loop

Depends on PR https://github.com/cloudfoundry/uaa/pull/2932

adrianhoelzl-sap commented on jobs/uaa/spec:
```diff
@@ -451,6 +451,9 @@ properties:
     description: "This flag enables the origin check in SCIM. Otherwise, the assignments of users to an origin are not validated."
     default: false
+  login.allowOriginLoop:
+    description: "This flag enables the loop over all origin of a certain type during login, e.g. all SAML or OIDC providers in case of such a logon. Otherwise, only index access is allowed."
```
```suggestion
    description: "This flag enables the loop over all origins of a certain type during login, e.g. all SAML or OIDC providers in case of such a logon. Otherwise, only index access is allowed."
```

strehle committed:
Update jobs/uaa/spec

Co-authored-by: Adrian Hölzl <102049638+adrianhoelzl-sap@users.noreply.github.com>

Tallicia commented on jobs/uaa/spec:
```diff
@@ -451,6 +451,9 @@ properties:
     description: "This flag enables the origin check in SCIM. Otherwise, the assignments of users to an origin are not validated."
     default: false
+  login.allowOriginLoop:
+    description: "This flag enables the loop over all origins of a certain type during login, e.g. all SAML or OIDC providers in case of such a logon. Otherwise, only index access is allowed."
```
What does "only index access is allowed" mean?

strehle replied:
It means we do only a search by external Id 
 -> SAML using entityID
 -> OIDC using issuer

The sentence maybe is too technical but it means, that we do not loop over all IdPs anymore.
The loop is for systems where you have 1000+ IdPs a problem. Currently we do a single lookup with external Key but do in addition the loop as before, simply because the externalKey is not forced to be there.

In SAP systems we ensure with updates to IdPs that the externalKey is available, because each Update creates the new field. After that we will set the toogle here and then we only want have a "select single" instead of "select *"

If you have better description, please do a commit to this PR

Tallicia reviewed (commented):
Please clarify in the spec based on the comment there.

strehle committed:
Update description

Use a more generic description.
This flag should be used later also in /identity-providers GET
to filter this and not return all IdPs
